travis-ci/terraform-config
# Replace NAT instances with Cloud NAT

## What is the problem that this PR is trying to fix?
The current NAT gateway is custom setup that with [instances as a NAT gateways](https://cloud.google.com/vpc/docs/special-configurations#natgateway) that work together with a [nat-conntracker](https://github.com/travis-ci/nat-conntracker/) run on Heroku.

This setup doesn't scale easily, is error prone and hard to keep in up-to-date.

## What approach did you choose and why?
Replace the whole setup with [Google's Cloud NAT](https://cloud.google.com/nat/docs/overview).

It provides us with a managed NAT gateway on regional network level. As it is managed, we can rely on the knowledge at Google to scale and update the NAT systems.

## How can you test this?
Install it on staging, see if jobs network is traversing through the NAT, great success.
For production, we should drain the workers before replacing the NATs, one project at a time.

## What feedback would you like, if any?
Sure, if you know more about Cloud NAT, I'd love to learn about it.
